src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -62,33 +57,26 @@ CHIP_ERROR TransferInit::Pack(System::PacketBuffer * aBuffer)
     if (mSupportsAsync)
         ptcByte |= ASYNC_MASK;
 
-    bbuf.Put(ptcByte);
-    VerifyOrExit(bbuf.Fit(), err = CHIP_ERROR_BUFFER_TOO_SMALL);
-
     if (mDefLen)
         rangeCtl |= DEFLEN_MASK;
     if (mStartOffset)
         rangeCtl |= START_OFFSET_MASK;
     if (mWideRange)
```

Comment:
Shouldn't the widerange choice be auto-computed based on the scale of the input values, rather than manually computed?

Reply:
I'm not sure what you mean here. I've chosen to represent the variable-length fields as uint64_t. If widerange is false, then the uint64_t fields will be truncated to uint32_t. What alternative are you suggesting?

Reply:
Discussed offline: will replace widerange boolean with inferring value from startoffset and maxlength fields. Same with mDefLen